drivers/ethernet/eth_stm32_hal.c

```diff
@@ -98,15 +227,16 @@ static int eth_tx(struct device *dev, struct net_pkt *pkt)
 	ETH_HandleTypeDef *heth;
 	uint8_t *dma_buffer;
 	int res;
-	uint16_t total_len;
+	size_t total_len;
 	__IO ETH_DMADescTypeDef *dma_tx_desc;
+	HAL_StatusTypeDef hal_ret = HAL_OK;
 
 	__ASSERT_NO_MSG(pkt != NULL);
 	__ASSERT_NO_MSG(pkt->frags != NULL);
 	__ASSERT_NO_MSG(dev != NULL);
 	__ASSERT_NO_MSG(dev_data != NULL);
 
-	heth = &dev_data->heth;
+	heth = &(dev_data->heth);
```

Comment:
the original code is just fine, please remove () here

Reply:
done